dashboard/app/helpers/levels_helper.rb

```diff
@@ -41,8 +41,16 @@ def set_videos_and_blocks_and_callouts_and_instructions
     select_and_track_autoplay_video
 
     if @level.is_a? Blockly
-      @toolbox_blocks = @toolbox_blocks || @level.toolbox_blocks
-      @start_blocks = initial_blocks(current_user, @level) || @start_blocks || @level.start_blocks
+      @toolbox_blocks =
+        @toolbox_blocks ||
+        @level.try(:project_template_level).try(:toolbox_blocks) ||
+        @level.toolbox_blocks
+
+      @start_blocks =
+        initial_blocks(current_user, @level) || # check if this level inherits solution from previous level
+        @start_blocks ||
+        @level.try(:project_template_level).try(:start_blocks) ||
+        @level.start_blocks
```

Comment:
Ah, gotcha.  FWIW I didn't see the symmetry in my initial reading but it makes sense now.  No strong preference here, go with whatever you think is best.